Source/Particles/Resampling/Resampling.cpp

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2019-2020 Neil Zaim
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "Resampling.H"
+#include "LevelingThinning.H"
+
+Resampling::Resampling ()
+{
+    amrex::ParmParse pp("resampling_algorithm");
+    std::string resampling_algorithm_string = "leveling_thinning"; // default resampling algorithm
+    pp.query("type", resampling_algorithm_string);
+
+    if (resampling_algorithm_string.compare("leveling_thinning") == 0)
+    {
+        m_resampling_algorithm.reset (new LevelingThinning );
```

Comment:
If you want to avoid `[...].reset ( new [...] )` you should be able to use a new `C++14` feature:
```suggestion
        m_resampling_algorithm = std::make_unique<LevelingThinning>();
```